### DISCUSSION TITLE
FPU extension verification

### DISCUSSION THREAD

**Discussion by Giorgio122** [stnolting/neorv32#577] [Q&A] [answered]
Hi everyone!
I am using the neorv32 on a pynq-Z2 board and were able to test its functionalities using the test_setup_bootloader wrapper;
I was trying to extend the neorv to use FP instructions by following the documentations and enabling the corresponding extensions. However, when I try to launch the test program (floating_point_test) it fails due to some values mismatch between the expected and actual outcome of the computation. Moreover, in these cases the RTE reports illegal instruction errors. I tried to analyze the .asm file in order to understand which was the instruction reporting the error and it corresponded to a custom R3 instruction type, despite having enabled also the CPU extension covering these situation. Does anyone has had this type of problem? How can I solve this? Is there maybe something wrong during the compilation of the C source code into the executable file(I am compiling it using the "make USER_FLAGS+=-DRUN_CHECK clean_all exe" command )?

![fpuErrors](https://user-images.githubusercontent.com/66768576/231715179-65f130c7-087b-48a8-be63-6470beee662b.png)

Thanks for your help,
Giorgio

**Comment by stnolting** [comment, marked as answer]
Hey Giorgio!

I just checked the FPU tests on my default setup - and they are also failing 🙈
Seems like I have broken something in one of my last "cleanup" commits... I will investigate on this! Thank you very much for discovering this bug!

**Reply by stnolting** [reply]
I made a PR in #578 that should fix the FPU bug.
Thanks again! 👍

**Reply by Giorgio122** [reply]
Thank you so much!

**Reply by stnolting** [reply]
Thank you for finding this bug!
